commons/src/main/java/org/infinispan/commons/persistence/Store.java

```diff
@@ -0,0 +1,21 @@
+package org.infinispan.commons.persistence;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Store. An annotation for identifying a persistent store and explicitly stating some of its characteristics.
+ *
+ * @author Ryan Emerson
+ * @since 9.0
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+public @interface Store {
+   /**
+    * Whether the store can be shared amongst nodes in a distributed cache
+    */
+   boolean shared() default true;
```

Comment:
I kinda wonder if we want to put it as false by default.  Being shared has restrictions so I would think a user would have to understand those to mark it as shared.